psh/pshapp/pshapp.c

```diff
@@ -1073,7 +1073,7 @@ static int psh_run(int exitable, const char *console)
 	pshapp_common.cmdhist = cmdhist;
 
 	while (pgrp == tcgetpgrp(STDIN_FILENO)) {
-		write(STDOUT_FILENO, "\r\033[0J", 5);
+		write(STDOUT_FILENO, "\033[0J", 5);
```

Comment:
After the change we want to write 4 bytes (without the '\0' byte) so it should be:
```c
write(STDOUT_FILENO, "\033[0J", 4);
```

Also, there are other places where shell prompt overwrites last line:
lines 618, 654, 725 and 735 in `pshapp.c`

I think we could remove `\r` there too.

Reply:
Thank you, I've changed all occurrences